modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -279,6 +279,8 @@ const whitespaceCleaner = (root: AstNode, schema: Schema, settings: DomParserSet
   const nonEmptyElements = schema.getNonEmptyElements();
   const whitespaceElements = schema.getWhitespaceElements();
   const blockElements: Record<string, string> = extend(makeMap('script,style,head,html,body,title,meta,param'), schema.getBlockElements());
+  // NOTE: This is duplicated in ApplyFormat.ts
+  const textRootBlockElements: Record<string, string> = extend(makeMap('td,th,li,dt,dd,figcaption,caption,details,summary'), schema.getTextBlockElements());
```

Comment:
Instead of duplicating this can we make it a shared function or similar please?

Reply:
I can't quite think of a good place for it so not sure if you have any thoughts? Maybe a new file is needed. It would be good to integrate into the schema somehow but I pretty sure that would require a minor.

Reply:
Is there any reason it can't just be in `Schema.ts`? That seems to make the most sense to me. It could either be an API on schema or just a floating function outside the schema that's not exposed on the public API.

Reply:
I have made the change. Let me know if this is along the lines of what you were thinking